syntax = "proto3";

service Bidirectional {
  rpc SubscribeStateListener(Subscribe) returns (stream NewState) {}
  rpc UnsubscribeStateListener(Subscribe) returns (Ok) {}
  rpc SimpleRequest(Message) returns (Message) {}
  rpc UpdateAnswer(Message) returns (Ok) {}
  rpc UpdateState(NewState) returns (Ok) {}
}

message Ok {
  string state = 1;
}

message NewState {
  string state = 1;
}

message Subscribe {
  string id = 1;
}

message Unsubscribe {
  string id = 1;
}

message Message {
  string id = 1;
  string data = 2;
}
